autoconfigure/storage-elasticsearch-aws/src/main/java/zipkin/autoconfigure/storage/elasticsearch/aws/AWSSignatureVersion4.java

```diff
@@ -111,16 +111,18 @@
 
               // We only set a body-related message when it is Amazon's format
               StringBuilder message = new StringBuilder().append(req.path()).append(" failed: ");
-              String awsMessage = null;
-              try (InputStream stream = aggResp.content().toInputStream()) {
-                awsMessage = JSON.readTree(stream).path("message").textValue();
-              } catch (IOException e) {
-                // Ignore JSON parse failure.
-              } finally {
-                // toInputStream creates an additional reference instead of itself releasing content()
-                ReferenceCountUtil.safeRelease(aggResp.content());
+              if (!aggResp.content().isEmpty()) {
+                String awsMessage = null;
+                try (InputStream stream = aggResp.content().toInputStream()) {
+                  awsMessage = JSON.readTree(stream).path("message").textValue();
+                } catch (IOException e) {
+                  // Ignore JSON parse failure.
+                } finally {
+                  // toInputStream creates an additional reference instead of itself releasing content()
+                  ReferenceCountUtil.safeRelease(aggResp.content());
```

Comment:
maybe we should check zipkin main code for this..

Reply:
@anuraaga is this legit? https://github.com/openzipkin/zipkin/pull/2743/commits/42a7929253d6137e78bae90fe9874e7dc30a4f83

I wonder if there's something we can add to `@After` to ensure all references that should have ended in zero are zero..